js/views/userPage/CategoryFilter.js

```diff
@@ -70,6 +74,10 @@ export default class extends BaseVw {
   }
 
   render() {
+    if (this.getState().categories.length === 1) {
```

Comment:
This should be checking if the length is zero, because "all" is added in the template. If the length is 1, you actually have 2 choices, "all" and the 1 category, and it shouldn't be disabled.

Reply:
I'm not sure disabling it is the right thing to do in any case. If the only option is All, it's automatically selected and the user can't change it, so it doesn't hurt anything.

Reply:
> This should be checking if the length is zero, because "all" is added in the template. If the length is 1, you actually have 2 choices, "all" and the 1 category, and it shouldn't be disabled.

I think J just followed what I did in the Category filter. My logic was probably that even though you have two choices (`All` and `Something`), they both evaluate to the same result set, so it didn't make sense to even allow the toggle.

> I'm not sure disabling it is the right thing to do in any case. If the only option is All, it's automatically selected and the user can't change it, so it doesn't hurt anything.

That's true. We could remove these checks if you want.

Reply:
You could have 1 listing with the "something" category, and 1 with no category on it. In that situation "all" gets both listings, and "something" gets only 1 listing.

So you need to keep both all and something as choices if there is only 1 category. Unless we also check to see if all the listings have the one category?